app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -48,8 +50,8 @@ def accessible_puppet_proxies(obj)
     list
   end
 
-  def domain_subnets
-    accessible_related_resource(@domain, :subnets)
+  def domain_subnets(type)
+    accessible_related_resource(@domain, :subnets, :query => {:type => type})
```

Comment:
Why not `accessible_related_resource(@domain, :subnets).where(:type=> type)` and then there is no need for changing the `accessible_related_resource` function?